presto-hive/src/main/java/io/prestosql/plugin/hive/HiveReaderProjectionsAdaptingRecordCursor.java

```diff
@@ -184,7 +184,7 @@ public Object getObject(int field)
         List<Integer> dereferences = channelMappings[field].getDereferenceSequence();
 
         if (dereferences.isEmpty()) {
-            return delegate.getSlice(inputFieldIndex);
+            return delegate.getObject(inputFieldIndex);
```

Comment:
Why did this change? If this was a bug in a previous commit, apply it there, instead.

Reply:
I think you forgot to address this comment.